docs/arch/adr-011-restricting-access-to-content-by-organisation.md

```diff
@@ -0,0 +1,84 @@
+
+Therefore, we need an interim solution that restricts access within Mainstream Publisher to a subset of the content held within it, acknowledging that how this works might change longer-term.
+
+In the immediate term, the requirement is to limit access, for non-GDS content designers, to that content which is managed by their own department. Signon does provide the organisation that a user belongs to, which Mainstream Publisher will use in order to make authorisation decisions when attempting to view and edit content. There are some cases where a content item would need to be managed by content designers from multiple organisations. 
+
+For now the access rules can be binary, with users either given full access to a piece of content (view/edit/publish), or no access.
+
+This ADS focuses on the technical solution to restrict access, using the data supplied by Signon about the user (i.e. the organisation to which they belong).
+
+### Option 1: Use the primary publishing organisation
+
+All content that is published on GOV.UK by Mainstream Publisher has a `primary_publishing_organisation` association set on it. This is currently always set to GDS at the point of publishing the content, but we could store this association as part of the content item in Mainstream Publisher, and set it to the relevant organisation. When determining whether a user should have access to a content item, it's Primary Publishing Organisation field would be compared with the user's organisation.
+
+#### Disadvantages
+
+- The Primary Publishing Organisation association already exists for a purpose, and is not intended as a means of access control.
+- A content item can only have a single Primary Publishing Organisation, so this solution would not enable content designers from different organisations to be able to see and edit the same content item.
+- Due to Primary Publishing Organisation being a one-to-one association with a content item, it doesn't really help us on the journey to supporting multiple-organisation access to a content item.
+
+#### Advantages
+
+- The Primary Publishing Organisation field is an existing field, and so this solution does not require introducing any new fields/concepts (though it is potentially changing the meaning of the association slightly by using it for access control).
+
+### Questions
+
+The Primary Publishing Organisation data flows through the Publishing API and into the content stored in the Content Store. Does it make sense to be changing this data to support access control within Mainstream Publisher (it introduces a level of coupling between two related, but not identical, concepts).
+
+### Option 2: Introducing a new field on content items
+
+A new field could be introduced to content items within Mainstream Publisher to store the organisation(s) whose members should have access to that content item.
```

Comment:
Maybe we could add a note to refer to [slack thread](https://gds.slack.com/archives/C051S690LGL/p1720451970243779) pointing that even though this solution may be confusing for future devs, hardcoded PPO is currently not exposed to users, so they would not be aware of this potential data duplication.

Reply:
I've added some text.